File: src/js/9._instancia_interna.js
// Clase con instancia interna

class Coche {

    constructor(marca, color) {
        /** @type {string} */
        this.brand = marca
        /** @type {string} */
        this.color = color
        /** @type {string} */
        this.isParked = false
    }

    info() {
        this.isParked = true
        console.log(`${this.brand} se ha estacionado`)

        console.log('Coches estacionados')
    }

    showParking() {
        this.info()
        // instancias de objetos de la misma clase
        const cocheVecino = new Coche('zenvo', 'rojo')
        const cocheVecinoTwo = new Coche('mini', 'blanco')
        const cocheVecinoThree = new Coche('mazda', 'naranja')

        cocheVecino.isParked = true
        cocheVecinoTwo.isParked = true 
        cocheVecinoThree.isParked = true
        /** @type {Coche[]} */
        let estacionamiento = [cocheVecino, cocheVecinoTwo, cocheVecinoThree]

        for (let auto of estacionamiento) {
            console.log(auto)
        }

        return estacionamiento
    }
}

const cocheMain = new Coche('porsche', 'silver')
// view
/** @type {Coche[]} */
let lugares = cocheMain.showParking()
console.table(lugares)

File: src/python/9._instancia_interna.py
# Clase con instancia interna

class Coche:
    def __init__(self, marca: str, color: str):
        self.brand = marca
        self.color = color
        self.parked = False

    def __repr__(self) -> str :
        estado = "estacionado" if self.parked else "en camino a estacionarse"
        return f"Coche ({self.brand}, {self.color}) - {estado}"

    def info(self) -> None :
        self.parked = True
        print(f' 🐸 {self.brand} se ha estacionado.')

        print('Coches estacionados')

    def show_parking(self) :
        self.info()
        # Instancia de objetos de la misma clase
        coche_vecino = Coche('Koenigsegg', 'blanco')
        coche_vecino_two = Coche('Rimac', 'azul')
        coche_vecino_three = Coche('Lamborghini', 'verde')

        coche_vecino.parked = True
        coche_vecino_two.parked = True
        coche_vecino_three.parked = True
        estacionamiento: list[Coche] = [self, coche_vecino, coche_vecino_two, coche_vecino_three]
        for coche in estacionamiento:
            print(f'    - {coche}')
        
        return estacionamiento


# main
if __name__ == "__main__":

    coche_main = Coche('McLaren', 'morado')
    lugares: list[Coche] = coche_main.show_parking()
